### PR TITLE
Update manifest and add a tutorial

### PR DESCRIPTION
**Pull request by arthurpaulino**
* Update the manifest
* Add a basic tutorial for Megaparsec

### CODE REVIEW

File: Examples/Tutorial.lean [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import Megaparsec.String
+
+/-!
```
Comment (cognivore) [inline Examples/Tutorial.lean:3]: It's a bit low-level, not sure if intended.

File: Examples/Tutorial.lean [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  return .num $ String.toNat! str
+
+/-- A symbol must start with a letter, followed by any other alphanumeric character -/
+def symP : P Grammar := do
+  let c : Char ← satisfy Char.isAlpha
+  let cs : List Char ← many' (satisfy Char.isAlphanum)
+  return .sym $ ⟨c :: cs⟩
+
+/--
+Let's call numbers and symbols by "atoms".
```
Comment (cognivore) [inline Examples/Tutorial.lean:45]: Not sure what do you mean with "call by"?

Extra "by"?

File: Examples/Tutorial.lean [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+We also consume blank characters before and 
+-/
+partial def grammarP : P Grammar := do
+  blanksP -- discard blank characters before anything
+  let x ← atomP <|> listP
+  return x
+
+partial def listP : P Grammar := do
+  discard $ single '('
```
Comment (cognivore) [inline Examples/Tutorial.lean:87]: I think that using between and sepEndBy like it is done in Lisp.lean is more instructive.

File: Examples/Tutorial.lean [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+--     (Grammar.cons (Grammar.sym "inv") (Grammar.cons (Grammar.num 2) (Grammar.nil)))
+--     (Grammar.cons (Grammar.num 3) (Grammar.nil)))
+
+/-- Now we can write the final function for parsing source code -/
+def parseGrammar (code : String) : Except String Grammar :=
+  match parse grammarP code with
+  | .left err => throw (toString err)
+  | .right g => return g
+
+#eval parseGrammar "(add 0 3)"
```
Comment (cognivore) [inline Examples/Tutorial.lean:111]: It's a good tutorial showing how to build parsers from ground up.
Thank you for taking time to write it and lay it out in a friendly manner.
I think we can add another tutorial showing higher level combinators based on this one.
The only change I would request is to mention that these parsers are built from very basic inbuilt capabilities of megaparsec.

**Review by cognivore — APPROVED** [review]
See single comments

**Review by ixahedron — APPROVED** [review]

**Pull request merged**